homeassistant/components/fan/__init__.py

```diff
@@ -379,16 +285,9 @@ async def async_oscillate(self, oscillating: bool):
     @property
     def is_on(self):
         """Return true if the entity is on."""
-        return self.speed not in [SPEED_OFF, None]
-
-    @property
-    def speed(self) -> str | None:
-        """Return the current speed."""
-        if preset_mode := self.preset_mode:
-            return preset_mode
-        if (percentage := self.percentage) is None:
-            return None
-        return self.percentage_to_speed(percentage)
+        return (
+            self.percentage is not None and self.percentage > 0
+        ) or self.preset_mode is not None
```

Comment:
Side note: We should return `None` if all state properties are None. That means state is unknown.

We can change this separately.